### PR TITLE
feat: schema library support

### LINKED ISSUE #1278: pubsub Subscription: should nack the message failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: 98cc01c867032847997484392fd75438eaf41317
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/c72cf75d-e01a-4277-becb-a23f75755e82), [Sponge](http://sponge2/c72cf75d-e01a-4277-becb-a23f75755e82)
status: failed
<details><summary>Test output</summary><br><pre>done() called multiple times in test <pubsub Subscription should nack the message> of file /workspace/build/system-test/pubsub.js
Error: done() called multiple times in test <pubsub Subscription should nack the message> of file /workspace/build/system-test/pubsub.js
    at /workspace/build/src/subscription.js:271:45
        -> /workspace/src/subscription.ts:354:41</pre></details>

### LINKED ISSUE #1287: pubsub Subscription: should ack the message failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: 226596fa8df5bf35ddab7ff7039df748883d3092
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/995d47e3-35c2-4f74-b404-8aa1de9fccbe), [Sponge](http://sponge2/995d47e3-35c2-4f74-b404-8aa1de9fccbe)
status: failed
<details><summary>Test output</summary><br><pre>done() called multiple times in test <pubsub Subscription should ack the message> of file /workspace/build/system-test/pubsub.js
Error: done() called multiple times in test <pubsub Subscription should ack the message> of file /workspace/build/system-test/pubsub.js
    at /workspace/build/src/subscription.js:271:45
        -> /workspace/src/subscription.ts:354:41</pre></details>

### PR DESCRIPTION
**Pull request by feywind**
This is split from https://github.com/googleapis/nodejs-pubsub/pull/1262 because that one was getting too big. This PR covers only the actual library support, and test changes that happened around making that happen.

### CODE REVIEW

**Comment by feywind** [comment]
There are unit tests, but not system-tests yet. I _have_ tested the functionality by hand using samples in another PR, but I am hoping to do the system-tests in a separate PR to try to keep it smaller.

File: src/pubsub.ts [hunk 1]
```diff
@@ -300,6 +318,7 @@ export class PubSub {
     if (this.isOpen) {
       this.isOpen = false;
       this.closeAllClients_()
+        .then(() => this.schemaClient?.close())
```
Comment (alexander-fenster) [inline src/pubsub.ts:321]: Just wondering why the `schemaClient` should be closed separately and not as a part of `closeAllClients_()`.

**Review by alexander-fenster — APPROVED** [review]

File: src/pubsub.ts (hunk 1 above)
Comment (feywind) [reply src/pubsub.ts:321]: Ah, yeah, this is me breaking with tradition from the old code. Previously, the library would store a bunch of raw gax clients by service name, and then call into them directly. The new way that I'm using for schema is to create a standard gapic object and store that for usage on schema calls. So it's not part of the "all clients" collection.

Eventually I'd like to eliminate all of the old "all clients" bits and make things use gapic underneath. But for now I'm keeping it separate to make it obvious.

File: src/pubsub.ts [hunk 2]
```diff
@@ -677,6 +759,39 @@ export class PubSub {
...
+   * @example
+   * for await (const s of pubsub.listSchemas()) {
+   *   const moreInfo = await s.get();
+   * }
+   */
+  async *listSchemas(options?: CallOptions): AsyncIterable<Schema> {
+    const client = await this.getSchemaClient_();
+    const query = {
+      parent: this.name,
+      view: google.pubsub.v1.SchemaView.BASIC,
```
Comment (hongalex) [inline src/pubsub.ts:788]: Is there a way for users to pass in a different schema view here? We need to provide both Basic/Full views as options to be returned by `list`.

**Pull request merged**